provider/data_rescue_image.go

```diff
@@ -0,0 +1,90 @@
+package provider
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+
+	"github.com/src-d/terraform-provider-online-net/online"
+
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func dataRescueImage() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceResourceImageRead,
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:          schema.TypeString,
+				Optional:      true,
+				ForceNew:      true,
+				Description:   "exact name of the desired image",
+				ConflictsWith: []string{"name_filter"},
+			},
+			"name_filter": {
+				Type:          schema.TypeString,
+				Optional:      true,
+				ForceNew:      true,
+				Description:   "partial name of the desired image to filter with",
+				ConflictsWith: []string{"name"},
+			},
+			"server": {
+				Type:        schema.TypeInt,
+				Required:    true,
+				ForceNew:    true,
+				Description: "server for the desired image",
+			},
+			"image": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "the requested image",
+			},
+		},
+	}
+}
+
+func dataSourceResourceImageRead(d *schema.ResourceData, meta interface{}) error {
+	c := meta.(online.Client)
+
+	server := d.Get("server").(int)
+	nameInterface, hasName := d.GetOk("name")
+	nameFilterInterface, hasNameFilter := d.GetOk("name_filter")
+
+	if !hasName && !hasNameFilter {
+		return errors.New("Need either a name or a name_filter")
+	}
+
+	images, err := c.GetRescueImages(server)
+	if err != nil {
+		return err
+	}
+
+	selectedImage := ""
+
+	if hasName {
+		want := nameInterface.(string)
+		for _, image := range images {
+			if image == want {
+				selectedImage = image
+			}
+		}
+	}
+
+	if hasNameFilter {
+		want := nameFilterInterface.(string)
+		for _, image := range images {
+			if strings.Contains(image, want) {
+				selectedImage = image
```

Comment:
I understand you cannot break here as we may want the last matching image, but at least this behaviour should be documented somewhere

Reply:
Added in description